drivers/gpio/gpio_emul.c

```diff
@@ -92,6 +92,10 @@ struct gpio_emul_data {
 	gpio_port_pins_t interrupts;
 	/** Spinlock to synchronize accesses to driver data and config */
 	struct k_spinlock lock;
+	/** Is interrupt enabled for each pin */
+	gpio_port_pins_t enabled_interrupts;
+	/** Mutex to synchronize accesses to driver data and config */
+	struct k_mutex mu;
```

Comment:
It looks like there is a conflict here. The old locking mechanism (mutex mu), was replaced with a new one (spinlock lock) somewhat recently. Please update

Reply:
Thank you for catching this. Removed unused `k_mutex`, and I think changes in this file should work with the spin lock.